### PR TITLE
Fix compiler warnings. Dynamically allocate weights_cumsum instead of

### PR DESCRIPTION
**Pull request by notgwj**
relying on a variable length array.  Initialize a variable before use.

### CODE REVIEW

File: arena.c [hunk 1]
```diff
@@ -42,7 +42,11 @@ static void arena_weighted_mbind(void *arena, size_t arena_size,
 	 * the method for determining a hit on a weight i is when the generated
 	 * random number (modulo sum of weights) <= weights_cumsum[i]
 	 */
-	int64_t weights_cumsum[nr_weights];
+	int64_t *weights_cumsum = malloc(nr_weights*sizeof(int64_t));
```
Comment (seranian) [inline arena.c]: Please add the call to free() when you are done with the table.

File: permutation.h [hunk 2]
```diff
@@ -80,13 +80,19 @@ static inline void rng_init(unsigned thread_num)
         rng_buf = (char*)calloc(1, RNG_BUF_SIZE);
         rand_state = (struct random_data*)calloc(1, sizeof(struct random_data));
         assert(rand_state);
-        assert(initstate_r(thread_num, rng_buf, RNG_BUF_SIZE, rand_state) == 0);
+        if (initstate_r(thread_num, rng_buf, RNG_BUF_SIZE, rand_state) != 0) {
+                perror("initstate_r");
+                exit(1);
```
Comment (seranian) [inline permutation.h]: suggest use #include <err.h>
and then
err(1, "initstate_r");
Print the  as a string, adds \n, and exit with code 1. All in one.

File: permutation.h [hunk 3]
```diff
@@ -80,13 +80,19 @@ static inline void rng_init(unsigned thread_num)
...
+        }
 }
 
 static inline perm_t rng_int(perm_t limit)
 {
         int r = 0;
-        assert(random_r(rand_state, &r) == 0);
+        if (random_r(rand_state, &r) != 0) {
+                perror("random_r");
+                exit(1);
```
Comment (seranian) [inline permutation.h]: ditto here

File: permutation.h (hunk 2 above)
Comment (notgwj) [inline permutation.h]: The rest of the code does it this way.  I'd prefer to be consistent with that.  I'm also not sure if err() is available everywhere.  The manpage says, "These functions are nonstandard BSD extensions."

File: arena.c (hunk 1 above)
Comment (notgwj) [inline arena.c]: Added in 10623ca9b47cf36380431bf10dc70709dcb3bc73.

**Comment by seranian** [comment]
On Fri, Apr 28, 2017 at 7:31 AM, notgwj <notifications@github.com> wrote:

> *@notgwj* commented on this pull request.
> ------------------------------
>
> In arena.c
> <https://github.com/google/multichase/pull/12#discussion_r113943634>:
>
> > @@ -42,7 +42,11 @@ static void arena_weighted_mbind(void *arena, size_t arena_size,
>  	 * the method for determining a hit on a weight i is when the generated
>  	 * random number (modulo sum of weights) <= weights_cumsum[i]
>  	 */
> -	int64_t weights_cumsum[nr_weights];
> +	int64_t *weights_cumsum = malloc(nr_weights*sizeof(int64_t));
>
> Added in 10623ca
> <https://github.com/google/multichase/commit/10623ca9b47cf36380431bf10dc70709dcb3bc73>
> .
>

Ok, submit the pull request for this one as well, then.
Merged your previous change now.

**Comment by notgwj** [comment]
10623ca was another commit in this branch that you already merged.  :-)

**Pull request merged**